[4.1] RavenDB-11452

ppekrol opened the pull request:
- AsyncDocumentSession.LoadAsync should return default(T) when id is null to match sync session behavior
- added various null argument checks
- synchronized AsyncDocumentSession.Load and DocumentSession.Load implementations

ayende commented on src/Raven.Client/Documents/Session/AsyncDocumentSession.Load.cs:
```diff
@@ -28,25 +30,35 @@ public partial class AsyncDocumentSession
 
-        public Task<Dictionary<string, T>> LoadAsync<T>(IEnumerable<string> ids, Action<IIncludeBuilder<T>> includes, CancellationToken token = default(CancellationToken))
+        /// <inheritdoc />
+        public Task<Dictionary<string, T>> LoadAsync<T>(IEnumerable<string> ids, Action<IIncludeBuilder<T>> includes, CancellationToken token = default)
```
Need to check `ids` for null

ppekrol committed:
RavenDB-11452
- AsyncDocumentSession.LoadAsync should return default(T) when id is null to match sync session behavior
- added various null argument checks
- synchronized AsyncDocumentSession.Load and DocumentSession.Load implementations
